discovery-service/src/main/java/org/zowe/apiml/discovery/ApimlInstanceRegistry.java

```diff
@@ -257,6 +260,52 @@ public void register(InstanceInfo info, final boolean isReplication) {
         }
     }
 
+    /**
+     * Validates that the service identifiers in the {@link InstanceInfo} are conformant and mutually consistent.
+     * The appName and vidAddress must not be null or empty. Both must comply with RFC 952 and RFC 1123.
+     * Only lowercase letters, digits, and hyphens allowed, must not start or end with a hyphen, and must not exceed 63 characters.
+     * The instanceId must follow the format 'hostname:serviceId:port'.
+     * The serviceId extracted from the instanceId must match both appName and vipAddress.
+     * If any of these checks fail, the method will throw an exception and the registration is rejected.
+     * @param info the instance info
+     * @throws MetadataValidationException exception if any service identifier is invalid or inconsistent
+     */
+    private void isServiceIdConformant(InstanceInfo info) {
+        String instanceId = info.getInstanceId();
+        String appName = info.getAppName().toLowerCase();
+        String vipAddress = info.getVIPAddress();
```

Comment:
```suggestion
        String vipAddress = StringUtils.lowerCase(info.getVIPAddress());
```
It could not passed regex below, or is it intentianal?

Actually, are we sure vipAdress is required?

Reply:
yeah it should be lower case, my bad. regarding whether the vipaddress is required, that should automatically calculated, so probably it's worthy to check only for application name